kernel/getset.c

```diff
@@ -170,3 +170,12 @@ int_t sys_capset(addr_t header_addr, addr_t data_addr) {
     STRACE("capset(%#x, %#x)", header_addr, data_addr);
     return 0;
 }
+
+// minimal version according to Linux sys/personality.h
+dword_t sys_personality(dword_t pers) {
+    if (pers == 0xffffffff)  // get personality, return default (Linux)
+        return 0x00000000;
+    if (pers == 0x00000000)  // set personality to Linux
+        return 0x00000000;
+    return -1;  // otherwise return error
```

Comment:
The kernel doesn't just return -1 on error, it returns a negative version of the errno. You can find these in kernel/errno.h.

Reply:
Allright - error should be _EINVAL and I've also corrected the return type to a signed int. BTW, on usual Linux (Arch Linux, x86_64), just any input works and sets personality to some value, so if something still fails with the syscall one could extend it later. Example:

```C
#include <sys/personality.h>
#include <stdio.h>

void main() {
  printf("testing personalities...\n");
  for (unsigned long i=0x00; i<=0xff; i++) {
    int ret = personality(i);
    printf("previous: %d, current: %d\n", ret, i);
  }
  printf("current: %d\n", personality(0xffffffff));
}
```